src/main/java/com/thealgorithms/datastructures/trees/LevelOrderTraversal.java

```diff
@@ -1,58 +1,46 @@
 package com.thealgorithms.datastructures.trees;
 
-public class LevelOrderTraversal {
-
-    class Node {
-
-        int data;
-        Node left, right;
-
-        public Node(int item) {
-            data = item;
-            left = right = null;
-        }
-    }
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Queue;
 
-    // Root of the Binary Tree
-    Node root;
-
-    public LevelOrderTraversal(Node root) {
-        this.root = root;
-    }
-
-    /* function to print level order traversal of tree*/
-    void printLevelOrder() {
-        int h = height(root);
-        int i;
-        for (i = 1; i <= h; i++) {
-            printGivenLevel(root, i);
-        }
-    }
+/* Class to print Level Order Traversal */
+public class LevelOrderTraversal {
 
-    /* Compute the "height" of a tree -- the number of
-  nodes along the longest path from the root node
-  down to the farthest leaf node.*/
-    int height(Node root) {
+    /* Given a binary tree. Print its nodes in level order
+  using array for implementing queue  */
+    static List<List<Integer>> traverse(BinaryTree.Node root) {
         if (root == null) {
-            return 0;
-        } else {
-            /**
-             * Return the height of larger subtree
-             */
-            return Math.max(height(root.left), height(root.right)) + 1;
+            return List.of();
         }
-    }
 
-    /* Print nodes at the given level */
-    void printGivenLevel(Node root, int level) {
-        if (root == null) {
-            return;
-        }
-        if (level == 1) {
-            System.out.print(root.data + " ");
-        } else if (level > 1) {
-            printGivenLevel(root.left, level - 1);
-            printGivenLevel(root.right, level - 1);
+        List<List<Integer>> result = new ArrayList<>();
+
+        Queue<BinaryTree.Node> q = new LinkedList<>();
+        q.add(root);
+        while (!q.isEmpty()) {
+            int nodesOnLevel = q.size();
+            List<Integer> level = new LinkedList<>();
+            /* poll() removes the present head.
+               For more information on poll() visit
+               http://www.tutorialspoint.com/java/util/linkedlist_poll.htm */
```

Comment:
Please remove trivial comments, they just duplicate what's already clear from the source code

Reply:
The original code and the comments aren't mine, I'll fix this.